### PR TITLE
feat(apig): support new data source

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
support new data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
support new data source
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/apig" TESTARGS="-run TestAccGroupsDataSource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccGroupsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGroupsDataSource_basic
=== PAUSE TestAccGroupsDataSource_basic
=== CONT  TestAccGroupsDataSource_basic
--- PASS: TestAccGroupsDataSource_basic (472.51s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      472.553s

make testacc TEST="./huaweicloud/services/acceptance/apig" TESTARGS="-run TestAccGroupsDataSource_filterById"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccGroupsDataSource_filterById -timeout 360m -parallel 4
=== RUN   TestAccGroupsDataSource_filterById
=== PAUSE TestAccGroupsDataSource_filterById
=== CONT  TestAccGroupsDataSource_filterById
--- PASS: TestAccGroupsDataSource_filterById (581.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      581.901s

make testacc TEST="./huaweicloud/services/acceptance/apig" TESTARGS="-run TestAccGroupsDataSource_filterByPreciseSearch"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccGroupsDataSource_filterByPreciseSearch -timeout 360m -parallel 4
=== RUN   TestAccGroupsDataSource_filterByPreciseSearch
=== PAUSE TestAccGroupsDataSource_filterByPreciseSearch
=== CONT  TestAccGroupsDataSource_filterByPreciseSearch
--- PASS: TestAccGroupsDataSource_filterByPreciseSearch (483.95s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      483.992s

```

### CODE REVIEW

File: huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_groups_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		},
+	})
+}
+
+func testAccGroupsDataSource_base(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_apig_group" "test" {
+  name        = "test_%s"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_groups_test.go]:
```suggestion
  name        = "%[2]s"
```
the `rName` already has 'test' string, the format is `tf_test_xxxxx`.

Comment (jinyangyang222) [reply huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_groups_test.go]: fixed

File: huaweicloud/services/apig/data_source_huaweicloud_apig_group.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+)
+
+func DataSourceGroups() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceGroupsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/data_source_huaweicloud_apig_group.go:23]: it's better to add **Computed** attr

Comment (jinyangyang222) [reply huaweicloud/services/apig/data_source_huaweicloud_apig_group.go:23]: fixed.

**Pull request merged**